src/shared/KubectlDetector.cs

```diff
@@ -9,46 +9,48 @@ namespace Microsoft.Tye
 {
     internal class KubectlDetector
     {
-        public static KubectlDetector Instance { get; } = new KubectlDetector();
+        private static bool? _kubectlInstalled = null;
```

Comment:
Curious why the whole Lazy construction was here? I looked at all the use cases and it doesn't seem like we have any thread safety concerns. I've simplified it to just be a bool, let me know if there is a reason to use Lazy instead.

Reply:
I mean in general I think the best practice here is to use lazy, as that is thread safe.

What was the motivation for removing Lazy?

Reply:
To pass in the output context?

Reply:
Nah, I just removed it to simplify. I can add the lazy back but it felt unnecessary, this never gets called in parallel.

Reply:
ewww a "best practice"

My original reasoning for using lazy was to avoid thread starvation or running multiple kubectl processes if we ever *do* run it in parallel. It was easy enough to avoid ever being a problem, and usually stuff that does build/publish grows parallel execution before long.